### ISSUE TITLE
setting custom user-agent header has no effect

### ISSUE THREAD

**Issue by kares** [logstash-plugins/logstash-filter-elasticsearch#157]
This is a regression with the Elasticsearch(-transport) gem.

Versions 7.14 / 7.15 work as expected but 7.16 and 7.17 started failing the test in place:
```
Failures:

  1) LogStash::Filters::Elasticsearch user-agent header used by plugin client should sent the expect user-agent
     Failure/Error: expect(request.header['user-agent'][0]).to match(/logstash\/\d*\.\d*\.\d* \(OS=.*; JVM=.*\) logstash-filter-elasticsearch\/\d*\.\d*\.\d*/)
     
       expected "elasticsearch-ruby/7.16.1 (RUBY_VERSION: 9.2.20.0; linux x86_64; Manticore 0.8.0)" to match /logstash\/\d*\.\d*\.\d* \(OS=.*; JVM=.*\) logstash-filter-elasticsearch\/\d*\.\d*\.\d*/
       Diff:
       @@ -1 +1 @@
       -/logstash\/\d*\.\d*\.\d* \(OS=.*; JVM=.*\) logstash-filter-elasticsearch\/\d*\.\d*\.\d*/
       +"elasticsearch-ruby/7.16.1 (RUBY_VERSION: 9.2.20.0; linux x86_64; Manticore 0.8.0)"
       
     # ./spec/filters/elasticsearch_spec.rb:453:in `block in <main>'
     # /opt/local/rvm/gems/jruby-9.2.20.0@ls/gems/logstash-devutils-2.3.0-java/lib/logstash/devutils/rspec/spec_helper.rb:61:in `block in <main>'

```

Setting the header using `transport_options[:headers].merge!('user-agent' => "#{user_agent}")` no longer works
`::Elasticsearch::Client.new(hosts: hosts, transport_options: transport_options, transport_class: ::Elasticsearch::Transport::Transport::HTTP::Manticore)` with the Manticore adapter.

---

NOTE: tried a quick variation using `transport_options[:headers].merge!('User-Agent' => ...)` since that is what the [library uses](https://github.com/elastic/elasticsearch-ruby/blob/7.16/elasticsearch-transport/lib/elasticsearch/transport/transport/base.rb#L370) as a base on top of the user-agent fallback ... but not luck

**Comment by kares** [comment]
turns out `"user-agent"` override never worked in ES client, the plugin only worked since [`< 7.16` had a patch](https://github.com/logstash-plugins/logstash-filter-elasticsearch/blob/v3.10.0/lib/logstash/filters/elasticsearch/patches/_elasticsearch_transport_http_manticore.rb#L6) applied

**Comment by kares** [comment]
upstream issue: https://github.com/elastic/elasticsearch-ruby/issues/1684

**Comment by andsel** [comment]
Hey @kares since the upstream issue is resolved for `7.17`should we also close this as resolved?
